src/main/java/com/commercetools/sync/taxcategories/utils/TaxRatesUpdateActionUtils.java

```diff
@@ -84,16 +91,17 @@ private TaxRatesUpdateActionUtils() {
         @Nonnull final List<TaxRate> oldTaxRates,
         @Nonnull final List<TaxRateDraft> newTaxRatesDrafts) {
 
+        List<TaxRateDraft> newTaxRateDraftsCopy = new ArrayList<>(newTaxRatesDrafts);
```

Comment:
Here why do you clone a copy of List<TaxRateDraft>?

Reply:
To avoid the ones that are replaced will not be added again. It is edge cases for when country code is the same but states are changed.
Check the scenarios: 
- buildTaxRatesUpdateActions_WithNullOldState_ShouldReturnOnlyReplaceAction
- buildTaxRatesUpdateActions_WithRemovedState_ShouldReturnOnlyReplaceAction